src/screens/AddSiteScreen.js

```diff
@@ -70,6 +63,14 @@ class AddSiteScreen extends Component {
         }
     };
 
+    componentWillMount() {
+        this.props.checkIfSiteIsReadyForUpload();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.props.checkIfSiteIsReadyForUpload();
+    }
```

Comment:
Here's what I was referring to in the reducer... basically as soon as the component mounts (in case rehydration brought along a fully complete site form) or anytime any props change (since the addition of a single letter may make the form complete), run a check to see if the form is populated. If so, it'll show the `Submit` button.